### ISSUE TITLE
Bump Jackrabbit to the latest 2.14 release

### ISSUE THREAD

**Issue by AndyScherzinger** [nextcloud/android-library#48]
At the moment we are using 2.12.x which is still using http client 3.1 while the new release 2.14 bumped the version to 4.1, see http://jackrabbit.510166.n4.nabble.com/ANNOUNCE-Apache-Jackrabbit-2-14-0-released-td4666058.html

> The main changes in this release are:
> - Switch WebDAV client components to HttpClient 4.5.x (summary in JCR-2406)
> - Introduction of Commons-VFS Datastore implementation (JCR-3975)
> - Allow use of JDK7 features (JCR-4030) 

cc @mario @tobiasKaminsky

**Comment by mario** [comment]
I completely support you in working on this! :)

**Comment by AndyScherzinger** [comment]
> Warning:WARNING: Dependency org.apache.httpcomponents:httpclient:4.5.2 is ignored for debug as it may be conflicting with the internal version provided by Android.

This might be an issue though since older versions of Android ship the http client themselves which it might need to be repacked, but then that hasn't been done before (ever)

**Comment by mario** [comment]
Fork, change name of the lib, include :D

**Comment by AndyScherzinger** [comment]
Name is not enough
you need to change the packages, else you will have collisions and depend on the classloader to pick the one you want!

**Comment by mario** [comment]
I know. I was joking :P Do it :D

**Comment by AndyScherzinger** [comment]
alternatively we could look at https://github.com/lookfirst/sardine but that might need a bit of work to integrate too since it might (also) not be Android compatible

**Comment by mario** [comment]
Already looked into that - not Android compatible, and the Android-compatible fork is really really bad. 

Long term, I think we'll either write our own lib or extend the one used by DavDroid to suit us, but for now ... I really think we should integrate 2.14 if we can in any way.

**Comment by AndyScherzinger** [comment]
Not sure if that works, since we would have to do what's been mentioned above, repackaging the complete thing... Since the build will drop the http client (which is incompatible to 3.x) and this is likely breaking the whole thing :/

**Comment by AndyScherzinger** [comment]
installed https://github.com/nextcloud/android/tree/jackrabbit214 on my phone and it seems to work... will have to test this a little.

Anyone having a IPv6 Installation?

**Comment by rubdos** [comment]
> Anyone having a IPv6 Installation?

Yes, is it the right time to test this now?

**Comment by AndyScherzinger** [comment]
> Yes, is it the right time to test this now?

No, I just asked since I don't have an IPv6 setup, so I can't really work on this issue even if I wanted to...

**Comment by tobiasKaminsky** [comment]
This is being worked on right now: I switched get activity operation already to newest v2 system.
